core/shared/src/main/scala/zio/internal/SingleThreadedRingBuffer.scala

```diff
@@ -48,7 +48,10 @@ private[zio] final class SingleThreadedRingBuffer[A <: AnyRef](capacity: Int) {
     var i                    = 0
     var result: List[AnyRef] = Nil
     while (i < array.length) {
-      result ::= array(i)
+      // https://github.com/zio/zio/issues/4384: We might concurrently drop the last
+      // element of the Ringbuffer and create the reversed list. Therefore we omit
+      // Null elements from the result.
+      if (array(i) != null) result ::= array(i)
```

Comment:
```suggestion
      val elem = array(i)
      if (elem != null) result ::= elem
```
Sorry for yet another nit pick :smile_cat: 
This indexes the array only once -- whether it has a meaningful performance impact, I can't really tell :shrug: 
Regardless, maybe it could express the idea -- that this is always about the same `array(i)` -- even clearer.

Reply:
No objections. Might be inetresting to compare the differences for such a case - just out of curiosity.